src/sso/interfaces/authorization-url-options.interface.ts

```diff
@@ -1,6 +1,6 @@
 export interface AuthorizationURLOptions {
   domain?: string;
-  projectID: string;
+  clientID: string;
```

Comment:
Looks like the Clubhouse story is for adding `clientID` and keeping `projectID` (to maintain backward compatibility)?

Reply:
I added projectID back as a supported paramater and rewrote the type interfaces and the tests to reflect in my most recent commit @marktran

Reply:
second commit resolved failing semaphore check

Reply:
> Looks like the Clubhouse story is for adding `clientID` and keeping `projectID` (to maintain backward compatibility)?

@marktran Looking at this again, I think the intention is to replace `projectID` with `clientID`. The `client_id` column on a V2 Project corresponds to the `id` column on a V1 Project.

I believe we had said that we were going to make this a breaking change and bump the major versions of the SDKs, so that when consumers update they would replace `projectID` with `clientID`.

We're already making breaking changes by removing `project_id` and replacing it with `environment_id` (see #284 and #286).

Reply:
@maxdeviant I thought the`client_id` column on V2 Project was backfilled with the `id` column from a V1 Project for existing users but had a generated id in a new format for new users?

Do you know if this migration strategy is documented anywhere? I was under the impression that we could make all of this backwards compatible so current users wouldn't have to change anything.

Reply:
@marktran I'm not familiar with the specifics of the `client_id` for new V2 projects.

I'm trying to find where we discussed the migration strategy for the SDKs. I definitely recall us saying that it would be a major version bump, which would indicate that we were making breaking changes. I have a feeling this was something that was said verbally and perhaps wasn't written down...

Reply:
We should probably get clarification from @rjvani before proceeding on these SDK changes. These stories don't have enough context.

I think there are some implications to making this a breaking change:
- Friction for current users to upgrade.
- Some current users will hold off on upgrading and we'll be forced to maintain compatibility with old versions 
- We'll need to push people to upgrade and that'll ultimately slow us down

Reply:
@marktran and @maxdeviant Should I withdraw this PR and just sit on the other client_id tickets until Rohan gets back? Or does this apply to all current SDK tickets that include breaking changes?

Reply:
@loferris I think we can keep this open and hold off on Clubhouse stories that have potentially breaking changes.